src/Microsoft.DotNet.Build.Tasks/PrereleaseResolveNuGetPackageAssets.cs

```diff
@@ -244,6 +256,37 @@ private void ExecuteCore()
             ProduceContentAssets(lockFile);
         }
 
+        private void PopulatePackageFolders(JObject lockFile)
+        {
+            // If we explicitly were given a path, let's use that
+            if (!string.IsNullOrEmpty(NuGetPackagesDirectory))
+            {
+                _packageFolders.Add(NuGetPackagesDirectory);
+            }
+
+            // Newer versions of NuGet can now specify the final list of locations in the lock file
+            var packageFolders = lockFile["packageFolders"] as JObject;
+
+            if (packageFolders != null)
+            {
+                foreach (var packageFolder in packageFolders.Properties())
+                {
+                    _packageFolders.Add(packageFolder.Name);
+                }
+            }
+
+            // If we didn't have any folders, let's fall back to the environment variable or user profile
+            if (_packageFolders.Count == 0)
+            {
+                string packagesFolder = Environment.GetEnvironmentVariable("NUGET_PACKAGES");
```

Comment:
Curiosity: Does this code path get used?  I understand that you just moved this from `GetNuGetPackagesPath`, but environment variables make me nervous